File: benchmarks/scenarios/cross-session-persistence/replay/anma/src/domains/accounts/service.py
"""accounts public surface. Kept decoupled from billing (see prior decision)."""

def get_user(user_id: str) -> dict:
    return {"id": user_id, "name": "Ada"}

def account_summary(user_id: str, invoiced: int = 0) -> dict:
    u = get_user(user_id)
    return {**u, "invoiced": invoiced}

def monthly_report(user_id: str, invoiced: int = 0) -> dict:
    # stays decoupled: invoiced injected by the caller, per the recorded decision
    u = get_user(user_id)
    return {**u, "invoiced": invoiced}

File: benchmarks/scenarios/go-payments/replay/anma/domains/accounts/service.go
package accounts

// GetUser returns a user's basic info.
func GetUser(userID string) map[string]string {
	return map[string]string{"id": userID, "name": "Ada"}
}

// AccountSummary returns user info plus their total invoiced amount.
// `invoiced` is injected by the caller to keep accounts decoupled from billing.
func AccountSummary(userID string, invoiced int) map[string]any {
	u := GetUser(userID)
	return map[string]any{"id": u["id"], "name": u["name"], "invoiced": invoiced}
}
